Add rough draft for new blog post (#113)
pharmaR/pharmaR.github.io: content/posts/news/2024-08-08-riskassessment-new-release-updates.Rmd

1. Choose a `{riskmetric}` assessment for which you would like to create a rule.
2. Construct a function taking the assessment value as an argument and return a logical (`TRUE`/`FALSE`).
3. Choose the decision category to assign if the test returns `TRUE`.

When constructing rules, it is important to know the structure of the assessment. All `{riskmetric}` assessments can be found in the [assessment function catalog](https://pharmar.github.io/riskmetric/reference/pkg_assess.html#assessment-function-catalog) on their website. Additionally, all helper functions from `{riskmetric}` are available for use when constructing the rules allowing a lot of flexibility in how the assessment can be parsed.

The rules can easily be re-ordered and will be read in a linear fashion. If no rules result in assigning a decision to the package, the `ELSE` option can be used to assign a decision but defaults to "No Decision".

<br>

![](/blog/2024-08-08-riskassessment-new-release-updates_files/decision_automation.gif)

<br>

> ## New `'Function Explorer'` page

Author: ```suggestion
## New `'Function Explorer'` Module
```

> Our friends at GSK have contributed a major addition that warranted a brand new page under the `'Source Explorer'` tab. Previously, a user could browser all the source code using the `'File Explorer'`. Now users can browse the details of individual functions exported from the package. More specifically, users can easily browser the source code, help documentation, and test files that reference the function of interest. This page extends the value of the application for organizations with a hands-on review process, giving users the ability to explore the package contents at a more granular level.

Author: ```suggestion
Our friends at GSK have contributed a major addition that warranted a brand new page under the `'Source Explorer'` tab. Previously, a user could browse all the source code using the `'File Explorer'`. Now users can investigate the details of individual functions exported from the package. More specifically, users have curated access to the source code, help documentation, and test files that reference the function of interest. This page extends the value of the application for organizations with a hands-on review process, giving users the ability to explore the package contents at a more granular level.
```

> Below, you will see a selection inputs panel on the left and a viewer pane on the right. Simply choose a function you want to further explore and what type of files. As previously mentioned, the type of files include the following: 

Author: ```suggestion
Below, you will see a selection inputs panel on the left and a viewer pane on the right. To get started, simply choose a function name & file types you want to explore. As previously mentioned, the type of files include the following: 
```

> 1. Testing Files: All discovered test files in the `tests/` folder containing the function will be displayed.

Author: ```suggestion
1. **Testing Files**: All discovered test files in the `tests/` folder containing the function will be displayed.
```

> 2. R Source Code: All R files in the `R/` folder containing the function will be displayed. This includes all usages of the function, not just where it was defined.

Author: ```suggestion
2. **R Source Code**: All R files in the `R/` folder containing the function will be displayed. This includes all usages of the function, not just where it was defined.
```

> 3. Help Documentation: The Rd file containing the function will be rendered in HTML.

Author: ```suggestion
3. Help Documentation: The `.Rd` file containing the function will be rendered in HTML.
```

> 3. Help Documentation: The Rd file containing the function will be rendered in HTML.

Author: ```suggestion
3. **Help Documentation**: The Rd file containing the function will be rendered in HTML.
```

Some additional features:

* If multiple lines containing the function call are found, navigation arrows will appear below the viewer pane, enabling the user to quickly jump to each instance.

> * Commenting is available for both the Function Explorer and the Package Browser.

Author: ```suggestion
* User comment sections are available for both the Function Explorer and the Package Browser.
```

* The `{archive}` package is being used to read these files into memory which greatly increases the speed for the `'Source Explorer'` tab overall.

<br>

![](/blog/2024-08-08-riskassessment-new-release-updates_files/function_explorer.gif)

<br>

## Expanded Dependency Exploration

> Cards are now included providing a macroscopic view of the dependencies!

Author: ```suggestion
Summary cards are now displayed to provide a macro view of the package's dependencies!
```

> * Dependencies Uploaded: the number of package dependencies that have been uploaded to the app

Author: ```suggestion
* **Dependencies Uploaded**: the number of package dependencies that have been uploaded to the app
```

> * Type Summary: a breakdown of the dependency types (Depends, Imports, and LinkingTo)

Author: ```suggestion
* **Type Summary**: a breakdown of the dependency types (`Depends`, `Imports`, and `LinkingTo`)
```

> * Decision summary: a breakdown of the decisions applied to the dependency packages

Author: ```suggestion
* **Decision summary**: a breakdown of the decisions applied to the dependency packages
```

> * Base-R packages: a summary of which dependencies are considered part of base R (i.e. utils, stats, methods, etc.)

Author: ```suggestion
* **Base-R packages**: a summary of which dependencies are considered part of base R (i.e. `utils`, `stats`, `methods`, etc.)
```

Optionally the non-required packages from the 'Suggests' fields can also be included to give a more holistic view of a package's dependency structure. This can be helpful for organizations that require suggested packages to also be qualified in their GxP environment.

<br>

![](/blog/2024-08-08-riskassessment-new-release-updates_files/dependency_cards.png)

<br>

A new table has been added, which is populated with the reverse dependency packages that were previously uploaded to the application. Now, in the event that a package is being re-assessed/re-evaluated, a user can quickly look at the downstream packages effected by a change in classification.

<br>

![](/blog/2024-08-08-riskassessment-new-release-updates_files/reverse_dependencies.png)

<br>

## Miscellaneous Items

### Expanded `'About'` Tab

> * Contact Us: a page that directs users to report issues, ask questions, reach out to contribute, learn more about the R Validation Hub

Author: ```suggestion
* **Contact Us**: a page that directs users to report issues, ask questions, reach out to contribute, learn more about the R Validation Hub
```

> * Contributors and Companies: get to know contributors, past and present

Author: ```suggestion
* **Contributors and Companies**: get to know contributors, past and present
```
